# Happiness Report & CIA Factbook

Data from the Happiness Report combined with data from the CIA World Factbook to find possible correlations

## A big limitation here is that the Happiness Report is actually using data which occurs in the CIA World Factbook (such as GDP per capita or life expectancy), so correlations here are obvious. 

An analysis therefore is actually redundant. This is more fun but it will not provide any new insights.

In [8]:
import pandas as pd
import numpy as np
import plotly
import plotly.graph_objs as go
plotly.offline.init_notebook_mode(connected=True)

In [9]:
happiness = pd.read_csv('data/2017.csv', dtype={'Country': np.object}).drop(['Happiness.Score', 'Whisker.high', 'Whisker.low', 'Economy..GDP.per.Capita.'], axis=1)
happiness.head()

Country  Happiness.Rank    Family  Health..Life.Expectancy.   Freedom  \
0       Norway               1  1.533524                  0.796667  0.635423   
1      Denmark               2  1.551122                  0.792566  0.626007   
2      Iceland               3  1.610574                  0.833552  0.627163   
3  Switzerland               4  1.516912                  0.858131  0.620071   
4      Finland               5  1.540247                  0.809158  0.617951   

   Generosity  Trust..Government.Corruption.  Dystopia.Residual  
0    0.362012                       0.315964           2.277027  
1    0.355280                       0.400770           2.313707  
2    0.475540                       0.153527           2.322715  
3    0.290549                       0.367007           2.276716  
4    0.245483                       0.382612           2.430182

In [10]:
factbook = pd.read_csv('data/countries-of-the-world.csv', decimal=',', sep=',', dtype={'Country': np.object})
factbook['Country'] = factbook['Country'].str.strip() #removing the whitespaces after each country in factbook, they are all like 'Afghanistan ', 'Albania ' etc
factbook['Region'] = factbook['Region'].str.strip()
factbook.head()

Country                Region  Population  Area (sq. mi.)  \
0     Afghanistan  ASIA (EX. NEAR EAST)    31056997          647500   
1         Albania        EASTERN EUROPE     3581655           28748   
2         Algeria       NORTHERN AFRICA    32930091         2381740   
3  American Samoa               OCEANIA       57794             199   
4         Andorra        WESTERN EUROPE       71201             468   

   Pop. Density (per sq. mi.)  Coastline (coast/area ratio)  Net migration  \
0                        48.0                          0.00          23.06   
1                       124.6                          1.26          -4.93   
2                        13.8                          0.04          -0.39   
3                       290.4                         58.29         -20.71   
4                       152.1                          0.00           6.60   

   Infant mortality (per 1000 births)  GDP ($ per capita)  Literacy (%)  \
0                              163.07               700.0          36.0   
1                               21.52              4500.0          86.5   
2                               31.00              6000.0          70.0   
3                                9.27              8000.0          97.0   
4                                4.05             19000.0         100.0   

   Phones (per 1000)  Arable (%)  Crops (%)  Other (%)  Climate  Birthrate  \
0                3.2       12.13       0.22      87.65      1.0      46.60   
1               71.2       21.09       4.42      74.49      3.0      15.11   
2               78.1        3.22       0.25      96.53      1.0      17.14   
3              259.5       10.00      15.00      75.00      2.0      22.46   
4              497.2        2.22       0.00      97.78      3.0       8.71   

   Deathrate  Agriculture  Industry  Service  
0      20.34        0.380     0.240    0.380  
1       5.22        0.232     0.188    0.579  
2       4.61        0.101     0.600    0.298  
3       3.27          NaN       NaN      NaN  
4       6.25          NaN       NaN      NaN

## In order to combine both dataframes, I convert both dataframes with the pivot_table function first and concatenate then

In [11]:
factbook_pivot = factbook.pivot_table(columns=['Country'])
happiness_pivot = happiness.pivot_table(columns=['Country'])

In [12]:
data = pd.concat([factbook_pivot, happiness_pivot])#concatenating both pivot tables together
data = data.astype(object)
data = data.T
data = data.dropna(subset=['Happiness.Rank']) #delete all the rows/countries that don't appear in the happiness report
data_plot = data #for plotting later
data = data.dropna() #delete any rows that have no data
data.head()

Agriculture Arable (%) Area (sq. mi.) Birthrate Climate  \
Afghanistan        0.38      12.13         647500      46.6       1   
Albania           0.232      21.09          28748     15.11       3   
Algeria           0.101       3.22    2.38174e+06     17.14       1   
Argentina         0.095      12.31    2.76689e+06     16.73       3   
Armenia           0.239      17.55          29800     12.07       4   

            Coastline (coast/area ratio) Crops (%) Deathrate  \
Afghanistan                            0      0.22     20.34   
Albania                             1.26      4.42      5.22   
Algeria                             0.04      0.25      4.61   
Argentina                           0.18      0.48      7.55   
Armenia                                0       2.3      8.23   

            GDP ($ per capita) Industry              ...               \
Afghanistan                700     0.24              ...                
Albania                   4500    0.188              ...                
Algeria                   6000      0.6              ...                
Argentina                11200    0.358              ...                
Armenia                   3500    0.343              ...                

            Pop. Density (per sq. mi.)   Population Service Dystopia.Residual  \
Afghanistan                         48   3.1057e+07    0.38            2.1508   
Albania                          124.6  3.58166e+06   0.579           1.49044   
Algeria                           13.8  3.29301e+07   0.298            2.5676   
Argentina                         14.4  3.99218e+07   0.547           2.61401   
Armenia                           99.9  2.97637e+06   0.418            1.5215   

               Family   Freedom Generosity Happiness.Rank  \
Afghanistan  0.581543   0.10618   0.311871            141   
Albania      0.803685  0.381499   0.201313            109   
Algeria       1.14622  0.233336  0.0694366             53   
Argentina     1.44045  0.494519   0.109457             24   
Armenia       1.00748  0.198303  0.0834881            121   

            Health..Life.Expectancy. Trust..Government.Corruption.  
Afghanistan                 0.180747                     0.0611578  
Albania                      0.73116                     0.0398642  
Algeria                     0.617585                      0.146096  
Argentina                   0.695137                     0.0597399  
Armenia                     0.637524                     0.0266744  

[5 rows x 25 columns]

After preparing the data, there are 121 countries left which are in the Happiness Report AND have no data in the factbook missing

## Let's look for correlations using the pandas .corr function

In [13]:
Happiness_Correlations = data.astype(float).corr(method='pearson', min_periods=10)
Happiness_Correlations = Happiness_Correlations[['Happiness.Rank']]
Happiness_Correlations.abs().sort_values(by='Happiness.Rank', ascending=False)

Happiness.Rank
Happiness.Rank                            1.000000
Health..Life.Expectancy.                  0.782471
GDP ($ per capita)                        0.752629
Family                                    0.736655
Phones (per 1000)                         0.731553
Infant mortality (per 1000 births)        0.707118
Birthrate                                 0.679356
Agriculture                               0.676177
Literacy (%)                              0.634532
Freedom                                   0.588837
Service                                   0.533932
Dystopia.Residual                         0.531605
Deathrate                                 0.512018
Trust..Government.Corruption.             0.448762
Climate                                   0.292136
Coastline (coast/area ratio)              0.269944
Industry                                  0.204484
Net migration                             0.178845
Generosity                                0.172256
Area (sq. mi.)                            0.135032
Crops (%)                                 0.122786
Pop. Density (per sq. mi.)                0.108967
Other (%)                                 0.085852
Arable (%)                                0.060335
Population                                0.023964

We can see that the highest correlation is with GDP per capita (from the Happiness Report) followed by life expectancy (From the Happiness Report) and GDP per Capita again (from the Factbook)

Or in other words, the more money per capita, the more happy a country is. 

Let's show that graphically

## World Graphs

In [14]:
plotly.offline.iplot({
    "data": [go.Scatter(x=data['Happiness.Rank'], 
                        y=data['Health..Life.Expectancy.'], 
                        mode="markers", 
                        hovertext=data.index,
                        )
            ],
    "layout": go.Layout(title="World",
                        xaxis=dict(title='Happiness Rank'),
                        yaxis=dict(title='Life Expectancy')
                       )
})

In [15]:
plotly.offline.iplot({
    "data": [go.Scatter(x=data['Happiness.Rank'], 
                        y=data['GDP ($ per capita)'], 
                        mode="markers", 
                        hovertext=data.index,
                        )
            ],
    "layout": go.Layout(title="World",
                        xaxis=dict(title='Happiness Rank'),
                        yaxis=dict(title='GDP per capita ($)')
                       )
})

In [16]:
plotly.offline.iplot({
    "data": [go.Scatter(x=data['Happiness.Rank'], 
                        y=data['Family'], 
                        mode="markers", 
                        hovertext=data.index,
                        )
            ],
    "layout": go.Layout(title="World",
                        xaxis=dict(title='Happiness Rank'),
                        yaxis=dict(title='Family Factor')
                       )
})

# Regions

Since there are high similarities among the cultures in certain regions, I want to extract regions and look at them seperately. In the newest Happiness Report, there is f.i. a section detailed about South America.

I want to look especially at Europe and South America. 

## Europe

In [17]:
factbook.Region.unique()

array(['ASIA (EX. NEAR EAST)', 'EASTERN EUROPE', 'NORTHERN AFRICA',
       'OCEANIA', 'WESTERN EUROPE', 'SUB-SAHARAN AFRICA',
       'LATIN AMER. & CARIB', 'C.W. OF IND. STATES', 'NEAR EAST',
       'NORTHERN AMERICA', 'BALTICS'], dtype=object)

In [18]:
factbook_europe = factbook.loc[factbook['Region'].isin(['EASTERN EUROPE', 'WESTERN EUROPE', 'BALTICS'])]
factbook_europe.head()

Country          Region  Population  Area (sq. mi.)  \
1                Albania  EASTERN EUROPE     3581655           28748   
4                Andorra  WESTERN EUROPE       71201             468   
12               Austria  WESTERN EUROPE     8192880           83870   
19               Belgium  WESTERN EUROPE    10379067           30528   
25  Bosnia & Herzegovina  EASTERN EUROPE     4498976           51129   

    Pop. Density (per sq. mi.)  Coastline (coast/area ratio)  Net migration  \
1                        124.6                          1.26          -4.93   
4                        152.1                          0.00           6.60   
12                        97.7                          0.00           2.00   
19                       340.0                          0.22           1.23   
25                        88.0                          0.04           0.31   

    Infant mortality (per 1000 births)  GDP ($ per capita)  Literacy (%)  \
1                                21.52              4500.0          86.5   
4                                 4.05             19000.0         100.0   
12                                4.66             30000.0          98.0   
19                                4.68             29100.0          98.0   
25                               21.05              6100.0           NaN   

    Phones (per 1000)  Arable (%)  Crops (%)  Other (%)  Climate  Birthrate  \
1                71.2       21.09       4.42      74.49      3.0      15.11   
4               497.2        2.22       0.00      97.78      3.0       8.71   
12              452.2       16.91       0.86      82.23      3.0       8.74   
19              462.6       23.28       0.40      76.32      3.0      10.38   
25              215.4       13.60       2.96      83.44      4.0       8.77   

    Deathrate  Agriculture  Industry  Service  
1        5.22        0.232     0.188    0.579  
4        6.25          NaN       NaN      NaN  
12       9.76        0.018     0.304    0.678  
19      10.27        0.010     0.240    0.749  
25       8.27        0.142     0.308    0.550

In [19]:
factbook_pivot_europe = factbook_europe.pivot_table(columns=['Country'])
factbook_pivot_europe.head()

Country           Albania  Andorra    Austria   Belgium  Bosnia & Herzegovina  \
Agriculture         0.232      NaN      0.018      0.01                 0.142   
Arable (%)         21.090     2.22     16.910     23.28                13.600   
Area (sq. mi.)  28748.000   468.00  83870.000  30528.00             51129.000   
Birthrate          15.110     8.71      8.740     10.38                 8.770   
Climate             3.000     3.00      3.000      3.00                 4.000   

Country           Bulgaria   Croatia  Czech Republic    Denmark   Estonia  \
Agriculture          0.093      0.07           0.034      0.018      0.04   
Arable (%)          40.020     26.09          39.800     54.020     16.04   
Area (sq. mi.)  110910.000  56542.00       78866.000  43094.000  45226.00   
Birthrate            9.650      9.61           9.020     11.130     10.04   
Climate              3.000       NaN           3.000      3.000      3.00   

Country              ...         Portugal     Romania  San Marino     Serbia  \
Agriculture          ...            0.053       0.101         NaN      0.166   
Arable (%)           ...           21.750      40.820       16.67     33.350   
Area (sq. mi.)       ...        92391.000  237500.000       61.00  88361.000   
Birthrate            ...           10.720      10.700       10.02        NaN   
Climate              ...            3.000       3.000         NaN        NaN   

Country          Slovakia   Slovenia      Spain      Sweden  Switzerland  \
Agriculture         0.035      0.028       0.04       0.011        0.015   
Arable (%)         30.160      8.600      26.07       6.540       10.420   
Area (sq. mi.)  48845.000  20273.000  504782.00  449964.000    41290.000   
Birthrate          10.650      8.980      10.06      10.270        9.710   
Climate             3.000        NaN       3.00       3.000        3.000   

Country         United Kingdom  
Agriculture              0.005  
Arable (%)              23.460  
Area (sq. mi.)      244820.000  
Birthrate               10.710  
Climate                  3.000  

[5 rows x 43 columns]

In [20]:
data_europe = pd.concat([factbook_pivot_europe, happiness_pivot]).astype(object).T.dropna(subset=['Happiness.Rank']).dropna() #concatenating both pivot tables together
data_europe

Agriculture Arable (%) Area (sq. mi.) Birthrate Climate  \
Albania              0.232      21.09          28748     15.11       3   
Austria              0.018      16.91          83870      8.74       3   
Belgium               0.01      23.28          30528     10.38       3   
Bulgaria             0.093      40.02         110910      9.65       3   
Czech Republic       0.034       39.8          78866      9.02       3   
Denmark              0.018      54.02          43094     11.13       3   
Estonia               0.04      16.04          45226     10.04       3   
Finland              0.028       7.19         338145     10.45       3   
France               0.022      33.53         547030     11.99       4   
Germany              0.009      33.85         357021      8.25       3   
Greece               0.054       21.1         131940      9.68       3   
Hungary              0.037      50.09          93030      9.72       3   
Iceland              0.086       0.07         103000     13.64       3   
Ireland               0.05       15.2          70280     14.45       3   
Latvia                0.04      29.67          64589      9.24       3   
Netherlands          0.021      26.71          41526      10.9       3   
Norway               0.021       2.87         323802     11.46       3   
Poland                0.05      45.91         312685      9.85       3   
Portugal             0.053      21.75          92391     10.72       3   
Romania              0.101      40.82         237500      10.7       3   
Spain                 0.04      26.07         504782     10.06       3   
Sweden               0.011       6.54         449964     10.27       3   
Switzerland          0.015      10.42          41290      9.71       3   
United Kingdom       0.005      23.46         244820     10.71       3   

               Coastline (coast/area ratio) Crops (%) Deathrate  \
Albania                                1.26      4.42      5.22   
Austria                                   0      0.86      9.76   
Belgium                                0.22       0.4     10.27   
Bulgaria                               0.32      1.92     14.27   
Czech Republic                            0      3.05     10.59   
Denmark                               16.97      0.19     10.36   
Estonia                                8.39      0.45     13.25   
Finland                                0.37      0.03      9.86   
France                                 0.63      2.07      9.14   
Germany                                0.67      0.59     10.62   
Greece                                10.37      8.78     10.24   
Hungary                                   0      2.06     13.11   
Iceland                                4.83         0      6.72   
Ireland                                2.06      0.03      7.82   
Latvia                                 0.82      0.47     13.66   
Netherlands                            1.09      0.97      8.68   
Norway                                 7.77         0       9.4   
Poland                                 0.16      1.12      9.89   
Portugal                               1.94      7.81      10.5   
Romania                                0.09      2.25     11.77   
Spain                                  0.98      9.87      9.72   
Sweden                                 0.72      0.01     10.31   
Switzerland                               0      0.61      8.49   
United Kingdom                         5.08      0.21     10.13   

               GDP ($ per capita) Industry              ...               \
Albania                      4500    0.188              ...                
Austria                     30000    0.304              ...                
Belgium                     29100     0.24              ...                
Bulgaria                     7600    0.304              ...                
Czech Republic              15700    0.393              ...                
Denmark                     3110

In [21]:
Happiness_Correlations_Europe = data_europe.astype(float).corr(method='pearson', min_periods=10)
Happiness_Correlations_Europe = Happiness_Correlations_Europe[['Happiness.Rank']]
Happiness_Correlations_Europe = Happiness_Correlations_Europe.abs().sort_values(by='Happiness.Rank', ascending=False)
Happiness_Correlations_Europe

Happiness.Rank
Happiness.Rank                            1.000000
GDP ($ per capita)                        0.839920
Dystopia.Residual                         0.816807
Trust..Government.Corruption.             0.796182
Freedom                                   0.791811
Generosity                                0.727384
Family                                    0.700464
Infant mortality (per 1000 births)        0.659755
Agriculture                               0.650629
Net migration                             0.639366
Phones (per 1000)                         0.622235
Literacy (%)                              0.575674
Health..Life.Expectancy.                  0.567153
Crops (%)                                 0.548443
Other (%)                                 0.410468
Arable (%)                                0.321050
Deathrate                                 0.318103
Service                                   0.233861
Area (sq. mi.)                            0.228174
Industry                                  0.190930
Pop. Density (per sq. mi.)                0.181555
Population                                0.133242
Coastline (coast/area ratio)              0.078502
Birthrate                                 0.037855
Climate                                   0.036076

Interesting to see that in Europe, the Life Expectancy is much lower related to happiness than worldwide. Possible reason might be that the life expectancy is fortunately so high in Europe that it doesn't matter anymore?

## Europe Graphs

In [22]:
plotly.offline.iplot({
    "data": [go.Scatter(x=data_europe['Happiness.Rank'], 
                        y=data_europe['GDP ($ per capita)'], 
                        mode="markers", 
                        hovertext=data_europe.index,
                        )
            ],
    "layout": go.Layout(title="Europe",
                        xaxis=dict(title='Happiness Rank'),
                        yaxis=dict(title='GDP per capita ($)')
                       )
})

In [23]:
plotly.offline.iplot({
    "data": [go.Scatter(x=data_europe['Happiness.Rank'], 
                        y=data_europe['Dystopia.Residual'], 
                        mode="markers", 
                        hovertext=data_europe.index,
                        )
            ],
    "layout": go.Layout(title="Europe",
                        xaxis=dict(title='Happiness Rank'),
                        yaxis=dict(title='Dystopia Residual')
                       )
})

In [24]:
plotly.offline.iplot({
    "data": [go.Scatter(x=data_europe['Happiness.Rank'], 
                        y=data_europe['Trust..Government.Corruption.'], 
                        mode="markers", 
                        hovertext=data_europe.index,
                        )
            ],
    "layout": go.Layout(title="Europe",
                        xaxis=dict(title='Happiness Rank'),
                        yaxis=dict(title='Corruption')
                       )
})

## Latin America

In [25]:
factbook.Region.unique()

array(['ASIA (EX. NEAR EAST)', 'EASTERN EUROPE', 'NORTHERN AFRICA',
       'OCEANIA', 'WESTERN EUROPE', 'SUB-SAHARAN AFRICA',
       'LATIN AMER. & CARIB', 'C.W. OF IND. STATES', 'NEAR EAST',
       'NORTHERN AMERICA', 'BALTICS'], dtype=object)

In [26]:
factbook_latinamerica = factbook.loc[factbook['Region'].isin(['LATIN AMER. & CARIB'])]
factbook_latinamerica.head()

Country               Region  Population  Area (sq. mi.)  \
6            Anguilla  LATIN AMER. & CARIB       13477             102   
7   Antigua & Barbuda  LATIN AMER. & CARIB       69108             443   
8           Argentina  LATIN AMER. & CARIB    39921833         2766890   
10              Aruba  LATIN AMER. & CARIB       71891             193   
14       Bahamas, The  LATIN AMER. & CARIB      303770           13940   

    Pop. Density (per sq. mi.)  Coastline (coast/area ratio)  Net migration  \
6                        132.1                         59.80          10.76   
7                        156.0                         34.54          -6.15   
8                         14.4                          0.18           0.61   
10                       372.5                         35.49           0.00   
14                        21.8                         25.41          -2.20   

    Infant mortality (per 1000 births)  GDP ($ per capita)  Literacy (%)  \
6                                21.03              8600.0          95.0   
7                                19.46             11000.0          89.0   
8                                15.18             11200.0          97.1   
10                                5.89             28000.0          97.0   
14                               25.21             16700.0          95.6   

    Phones (per 1000)  Arable (%)  Crops (%)  Other (%)  Climate  Birthrate  \
6               460.0        0.00       0.00     100.00      2.0      14.17   
7               549.9       18.18       4.55      77.27      2.0      16.93   
8               220.4       12.31       0.48      87.21      3.0      16.73   
10              516.1       10.53       0.00      89.47      2.0      11.03   
14              460.6        0.80       0.40      98.80      2.0      17.57   

    Deathrate  Agriculture  Industry  Service  
6        5.34        0.040     0.180    0.780  
7        5.37        0.038     0.220    0.743  
8        7.55        0.095     0.358    0.547  
10       6.68        0.004     0.333    0.663  
14       9.05        0.030     0.070    0.900

In [27]:
factbook_pivot_latinamerica = factbook_latinamerica.pivot_table(columns=['Country'])
factbook_pivot_latinamerica

Country                             Anguilla  Antigua & Barbuda     Argentina  \
Agriculture                             0.04              0.038  9.500000e-02   
Arable (%)                              0.00             18.180  1.231000e+01   
Area (sq. mi.)                        102.00            443.000  2.766890e+06   
Birthrate                              14.17             16.930  1.673000e+01   
Climate                                 2.00              2.000  3.000000e+00   
Coastline (coast/area ratio)           59.80             34.540  1.800000e-01   
Crops (%)                               0.00              4.550  4.800000e-01   
Deathrate                               5.34              5.370  7.550000e+00   
GDP ($ per capita)                   8600.00          11000.000  1.120000e+04   
Industry                                0.18              0.220  3.580000e-01   
Infant mortality (per 1000 births)     21.03             19.460  1.518000e+01   
Literacy (%)                           95.00             89.000  9.710000e+01   
Net migration                          10.76             -6.150  6.100000e-01   
Other (%)                             100.00             77.270  8.721000e+01   
Phones (per 1000)                     460.00            549.900  2.204000e+02   
Pop. Density (per sq. mi.)            132.10            156.000  1.440000e+01   
Population                          13477.00          69108.000  3.992183e+07   
Service                                 0.78              0.743  5.470000e-01   

Country                                 Aruba  Bahamas, The   Barbados  \
Agriculture                             0.004          0.03       0.06   
Arable (%)                             10.530          0.80      37.21   
Area (sq. mi.)                        193.000      13940.00     431.00   
Birthrate                              11.030         17.57      12.71   
Climate                                 2.000          2.00       2.00   
Coastline (coast/area ratio)           35.490         25.41      22.51   
Crops (%)                               0.000          0.40       2.33   
Deathrate                               6.680          9.05       8.67   
GDP ($ per capita)                  28000.000      16700.00   15700.00   
Industry                                0.333          0.07       0.16   
Infant mortality (per 1000 births)      5.890         25.21      12.50   
Literacy (%)                           97.000         95.60      97.40   
Net migration                           0.000         -2.20      -0.31   
Other (%)                              89.470         98.80      60.46   
Phones (per 1000)                     516.100        460.60     481.90   
Pop. Density (per sq. mi.)            372.500         21.80     649.50   
Population                          71891.000     303770.00  279912.00   
Service                                 0.663          0.90       0.78   

Country                                 Belize      Bolivia        Brazil  \
Agriculture                              0.142        0.128  8.400000e-02   
Arable (%)                               2.850        2.670  6.960000e+00   
Area (sq. mi.)                       22966.000  1098580.000  8.511965e+06   
Birthrate                               28.840       23.300  1.656000e+01   
Climate                                  2.000        1.500  2.000000e+00   
Coastline (coast/area ratio)             1.680        0.000  9.000000e-02   
Crops (%)                                1.710        0.190  9.000000e-01   
Deathrate                                5.720        7.530  6.170000e+00   
GDP ($ per capita)                    4900.000     2400.000  7.600000e+03   
Industry                                 0.152        0.352  4.000000e-01   
Infant mortality (per 1000 births)      25.690       53.110  2.961000e+01   
Literacy (%)                            94.100       87.200  8.640000e+01   
Net migration                            0.000      

In [28]:
data_latinamerica = pd.concat([factbook_pivot_latinamerica, happiness_pivot]).astype(object).T.dropna(subset=['Happiness.Rank']).dropna() #concatenating both pivot tables together
data_latinamerica.index

Index(['Argentina', 'Belize', 'Bolivia', 'Brazil', 'Chile', 'Colombia',
       'Costa Rica', 'Dominican Republic', 'Ecuador', 'El Salvador',
       'Guatemala', 'Haiti', 'Honduras', 'Jamaica', 'Mexico', 'Nicaragua',
       'Panama', 'Paraguay', 'Peru', 'Uruguay', 'Venezuela'],
      dtype='object')

In [29]:
Happiness_Correlations_LatinAmerica = data_latinamerica.astype(float).corr(method='pearson', min_periods=10)
Happiness_Correlations_LatinAmerica = Happiness_Correlations_LatinAmerica[['Happiness.Rank']]
Happiness_Correlations_LatinAmerica = Happiness_Correlations_LatinAmerica.abs().sort_values(by='Happiness.Rank', ascending=False)
Happiness_Correlations_LatinAmerica

Happiness.Rank
Happiness.Rank                            1.000000
Dystopia.Residual                         0.777581
Health..Life.Expectancy.                  0.730379
Phones (per 1000)                         0.701684
Infant mortality (per 1000 births)        0.678366
Freedom                                   0.662075
GDP ($ per capita)                        0.649866
Birthrate                                 0.645751
Literacy (%)                              0.583725
Family                                    0.582315
Pop. Density (per sq. mi.)                0.501317
Deathrate                                 0.472981
Crops (%)                                 0.472008
Coastline (coast/area ratio)              0.471057
Agriculture                               0.467188
Other (%)                                 0.459465
Arable (%)                                0.425573
Generosity                                0.413105
Population                                0.307295
Area (sq. mi.)                            0.304499
Climate                                   0.286705
Net migration                             0.267707
Service                                   0.194708
Industry                                  0.188163
Trust..Government.Corruption.             0.170875

## Latin America Graphs

In [30]:
plotly.offline.iplot({
    "data": [go.Scatter(x=data_latinamerica['Happiness.Rank'], 
                        y=data_latinamerica['Dystopia.Residual'], 
                        mode="markers", 
                        hovertext=data_latinamerica.index,
                        )
            ],
    "layout": go.Layout(title="Latin America",
                        xaxis=dict(title='Happiness Rank'),
                        yaxis=dict(title='Dystopia Residual')
                       )
})

In [31]:
plotly.offline.iplot({
    "data": [go.Scatter(x=data_latinamerica['Happiness.Rank'], 
                        y=data_latinamerica['Health..Life.Expectancy.'], 
                        mode="markers", 
                        hovertext=data_latinamerica.index,
                        )
            ],
    "layout": go.Layout(title="Latin America",
                        xaxis=dict(title='Happiness Rank'),
                        yaxis=dict(title='Life Expectancy')
                       )
})

In [32]:
plotly.offline.iplot({
    "data": [go.Scatter(x=data_latinamerica['Happiness.Rank'], 
                        y=data_latinamerica['Phones (per 1000)'], 
                        mode="markers", 
                        hovertext=data_latinamerica.index,
                        )
            ],
    "layout": go.Layout(title="Latin America",
                        xaxis=dict(title='Happiness Rank'),
                        yaxis=dict(title='Phones per 1000')
                       )
})

# World Maps

In [33]:
graph_world = [ dict(
        type = 'choropleth',
        locations = data_plot.index,
        locationmode = 'country names',
        z = data_plot['Happiness.Rank'],
        text = data_plot.index,
        colorscale = 'RdBu',
        marker = dict(
            line = dict (
                color = 'rgb(180,180,180)',
                width = 0.3
            ) ),
        hoverinfo = 'z+text'      
      ) ]

layout = dict(
    title = 'Happiness Rank',
    autosize=False,
    width=1000,
    height=900,
    geo = dict(
        showframe = False,
        showcoastlines = True,
        projection = dict(
            type = 'mercator'
        )
    )
)

plotly.offline.iplot(dict(data=graph_world, layout=layout))

In [34]:
graph_world_gdp = [ dict(
        type = 'choropleth',
        locations = data_plot.index,
        locationmode = 'country names',
        z = data_plot['GDP ($ per capita)'],
        text = data_plot.index,
        colorscale = 'Blues',
        autocolorscale = False,
        reversescale = True,
        marker = dict(
            line = dict (
                color = 'rgb(180,180,180)',
                width = 0.5
            ) ),
        hoverinfo = 'z+text'      
      ) ]

layout = dict(
    title = 'GDP per Capita (in $)',
    autosize=False,
    width=1000,
    height=900,
    geo = dict(
        showframe = False,
        showcoastlines = True,
        projection = dict(
            type = 'mercator'
        )
    )
)

plotly.offline.iplot(dict(data=graph_world_gdp, layout=layout))

In [35]:
graph_world_fam = [ dict(
        type = 'choropleth',
        locations = data_plot.index,
        locationmode = 'country names',
        z = data_plot['Family'],
        text = data_plot.index,
        colorscale = 'RdBu',
        autocolorscale = False,
        reversescale = True,
        marker = dict(
            line = dict (
                color = 'rgb(180,180,180)',
                width = 0.5
            ) ),
        hoverinfo = 'z+text'      
      ) ]

layout = dict(
    title = 'Family',
    autosize=False,
    width=1000,
    height=900,
    geo = dict(
        showframe = False,
        showcoastlines = True,
        projection = dict(
            type = 'mercator'
        )
    )
)

plotly.offline.iplot(dict(data=graph_world_fam, layout=layout))

In [36]:
graph_world_fre = [ dict(
        type = 'choropleth',
        locations = data_plot.index,
        locationmode = 'country names',
        z = data_plot['Freedom'],
        text = data_plot.index,
        colorscale = 'RdBu',
        autocolorscale = False,
        reversescale = True,
        marker = dict(
            line = dict (
                color = 'rgb(180,180,180)',
                width = 0.5
            ) ),
        hoverinfo = 'z+text'      
      ) ]

layout = dict(
    title = 'Freedom',
    autosize=False,
    width=1000,
    height=900,
    geo = dict(
        showframe = False,
        showcoastlines = True,
        projection = dict(
            type = 'mercator'
        )
    )
)

plotly.offline.iplot(dict(data=graph_world_fre, layout=layout))

In [37]:
graph_world_bir = [ dict(
        type = 'choropleth',
        locations = data_plot.index,
        locationmode = 'country names',
        z = data_plot['Birthrate'],
        text = data_plot.index,
        colorscale = 'Blues',
        autocolorscale = False,
        reversescale = True,
        marker = dict(
            line = dict (
                color = 'rgb(180,180,180)',
                width = 0.5
            ) ),
        hoverinfo = 'z+text'      
      ) ]

layout = dict(
    title = 'Birthrate',
    autosize=False,
    width=1000,
    height=900,
    geo = dict(
        showframe = False,
        showcoastlines = True,
        projection = dict(
            type = 'mercator'
        )
    )
)

plotly.offline.iplot(dict(data=graph_world_bir, layout=layout))